packages/dependency-extraction-webpack-plugin/README.md

```diff
@@ -0,0 +1,67 @@
+# Dependency Extraction Webpack Plugin
+
+This webpack plugin serves two purposes:
+
+- Externalize dependencies that are available as script dependencies on modern WordPress sites.
+- Add a JSON file for each entrypoint that declares the WordPress script dependencies for the
+  entrypoint.
+
+This allows JavaScript bundles produced by webpack to leverage WordPress style dependency sharing
+without an error-prone process of manually maintaining a dependency list.
+
+Consult the [webpack website](https://webpack.js.org) for additional information on webpack concepts.
+
+## Installation
+
+Install the module
+
+```bash
+npm install @wordpress/dependency-extraction-webpack-plugin --save-dev
+```
+
+## Usage
+
+### Webpack
+
+Use this plugin as you would other webpack plugins:
+
+```js
+// webpack.config.js
+const WordPressExternalDependenciesPlugin = require( '@wordpress/dependency-extraction-webpack-plugin' );
+
+module.exports = {
+  // …snip
+  plugins: [
+    new WordPressExternalDependenciesPlugin(),
+  ]
+}
+```
+
+Each entrypoint in the webpack bundle will include JSON file that declares the WordPress script dependencies that should be enqueued.
```

Comment:
We should explain what happens to the existing `externals` when they are provided as well.

Reply:
`externals` can still be used, I hope the documentation makes it clear (https://github.com/WordPress/gutenberg/pull/14869#discussion_r275461917).